src/classes/PMT_Payment_TEST.cls

```diff
@@ -837,6 +837,871 @@ private class PMT_Payment_TEST {
+            // success
+            System.assert(ex.getMessage().contains(System.Label.Opp_Amount_Not_Equal_Error), 'The opportunity was not updated, but received different error message.');
+        }
+    }
+
+    /* Test with Enforce Accounting Data Consistency Enabled */
+    /* @description Non Final Opportunity inserted with unpaid payments totaling < than opportunity Amount. Change Unpaid Payments to Paid. Amount should remain the same
+    */
+    @isTest
+    private static void oppNonFinalizedUpdatePaymentsToPaid() {
+        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c = true, Enforce_Accounting_Data_Consistency__c = true));
+        Account acc = [SELECT Id FROM Account LIMIT 1];
+
+        Test.startTest();
+
+        Opportunity opp = new Opportunity(
+                Name='committed opp',
+                Amount = 100,
+                AccountID = acc.Id,
+                CloseDate = System.today(),
+                npe01__Do_Not_Automatically_Create_Payment__c = true,
+                StageName = OPP_StageMappingUtil.TEST_COMMITTED_STAGE);
+        insert opp;
+
+        List<npe01__OppPayment__c> payments = new List<npe01__OppPayment__c> {
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = false,
+                        npe01__Payment_Amount__c = 40,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Scheduled_Date__c = System.today()
+                ),
+                new npe01__OppPayment__c(
+                        npe01__Paid__c = false,
+                        npe01__Payment_Amount__c = 40,
+                        npe01__Opportunity__c = opp.Id,
+                        npe01__Scheduled_Date__c = System.today()
+                )
+        };
+        insert payments;
+
+        // update payments to paid
+        for (npe01__OppPayment__c p : payments) {
+            p.npe01__Paid__c = true;
+            p.npe01__Payment_Date__c = System.today();
+        }
+        update payments;
+        
+        Test.stopTest();
+
+        // check the opp amount
+        opp = [select Id, Amount from Opportunity where Id = :opp.Id];
+        System.assertEquals(100, opp.Amount, 'The opportunity amount should have remained unchanged.');
+        
+    }
+
+
+    /* Test with Enforce Accounting Data Consistency Enabled */
+    /* @description Non Finalized Opportunity inserted with unpaid payments totaling > than opportunity Amount. 
+                    Change Unpaid Payments to Paid. Amount should increase to Sum of Paid Payments
+    */
+    @isTest
+    private static void oppNonFinalizedUpdateHigherPaymentsToPaid() {
+        UTIL_CustomSettingsFacade.getContactsSettingsForTests(new npe01__Contacts_And_Orgs_Settings__c (npe01__Payments_Enabled__c = true, Enforce_Accounting_Data_Consistency__c = true));
+        Account acc = [SELECT Id FROM Account LIMIT 1];
```

Comment:
Remind me to show you some of the NPSP test helper methods for creating data.  They can be useful to make test classes that can run multiple tests at once, which can reduce test execution times.